XmlSupportPkg/Library/XmlTreeLib/fasterxml/xml_fasterxml.c

```diff
@@ -4934,7 +4935,8 @@ RtlXmlCopyStringOut (
     // Two chars required
     //
     else if (Result.Character < 0x110000) {
-      if (((pwszWriteEnd + 2) <= pwszWriteEnd) && (pwszWriteCursor != NULL)) {
+      TempPtrEnd = (UINTN)pwszWriteEnd;
+      if (((TempPtrEnd + 2 * sizeof (*pwszWriteEnd)) <= TempPtrEnd) && (pwszWriteCursor != NULL)) {

```

Comment:
From an operator precedence perspective this is fine, but some parentheses would make the logic easier to follow.

```suggestion
      if (((TempPtrEnd + (2 * sizeof (*pwszWriteEnd))) <= TempPtrEnd) && (pwszWriteCursor != NULL)) {
```